sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -6229,6 +6229,71 @@ object functions {
     Bucket(Literal(numBuckets), e.expr)
   }
 
+  //////////////////////////////////////////////////////////////////////////////////////////////
+  // Predicates functions
+  //////////////////////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Returns `col2` if `col1` is null, or `col1` otherwise.
+   *
+   * @group predicates_funcs
+   * @since 3.5.0
+   */
+  def ifnull(col1: Column, col2: Column): Column = withExpr {
```

Comment:
the `replacement` in `ifnull`/`nvl`/`nvl2`/`nullif` is missing?

Reply:
According to my understanding, 'replacement: Expression' is only an internal Spark mechanism that exists to cooperate with 'RuntimeReplaceable' and is designed to reuse expressions. I guess it is actually not needed in our function.

Reply:
@zhengruifeng the default constructor with `replacement: Expression` will be replaced by rules. We only need consider the other constructor without `replacement`.

Reply:
oh, thanks for the explanation! @panbingkun @beliefer